### ISSUE TITLE
Support for Model Engine

### ISSUE THREAD

**Issue by ghost** [Arnuh/EchoPet#104]
Hey,

Was curious if you could add support for modelengine? Modelengine is only used alongside plugins such as Citizens 2 (for custom npcs) or MythicMobs for Custom bosses and mobs but can be used as a stand-alone plugin as well.

ModelEngine > [Their Github](https://github.com/Ticxo/Model-Engine-Wiki)

ModelEngine in action > [Creating Custom Mobs Plugin | Minecraft](https://www.youtube.com/watch?v=OaVdUaaNweE)

**Comment by ghost** [comment]
Upon second thought: considering modelengine uses multiple armorstands per limb (meaning more lag per pet) i was curious if it would be possible to also have support/an option for that singular armorstand pet model idea with expandability for allowing custommodeldata items in the headslot of each "armorstand pet" and mounting height. 

here's an example of how the config could look again:
![hc](https://user-images.githubusercontent.com/65664977/134271576-4e7ccda5-37e8-48b0-9844-17151047c3c7.png)

**Comment by Arnuh** [comment]
Does Model Engine not support doing models with 1 armor stand? I thought the 'limbs' were required to be added when modeled.

I can still add a PetData which is like "Custom Model" and when clicked opens up a ui with a bunch of customizable items that puts an item in the headslot of the current pet with the custommodeldata properties. Would be an alternative over using ModelEngine.

**Comment by ghost** [comment]
It does yeah i believe.

That second idea might be a decent one but i think it's hard to really then make "presets" of each custom modeled pet you might want that *seems* as though is already built into the pet plugin. Say indeed it is too difficult to do something like this i could always look into using mythic mobs i suppose (albeit it might be a little more complex) 

*edit* ->
just had to confirm, so yeah model engine would only use one armorstand.

in other words: totally fine to use the modelengine/echopet idea actually upon 2nd thought

**Comment by Arnuh** [comment]
After a decently large change to EchoPet, it now at supports a little bit of custom pets. Plugin currently automatically detects ModelEngine models and creates a custom pet for them with tab complete support. The actual entity it spawns the model on is a cow which is invisible & muted, not sure if a better option exists for that part.

It is probably best to make it so you have to input a list of model names in a config to prevent every model showing in the tab complete list & because Model Engine offers no way for you to listen to models being loaded.

![image](https://user-images.githubusercontent.com/2877803/134768630-61cd225c-728a-4731-8f4a-b700c29ccde0.png)

**Comment by ghost** [comment]
Ooo exciting stuff. Is the addon ready or is it already apart of the latest echopet jenkins jar outta curiosity?

**edit** : oh nvm i just saw the commits as unfinished in regards to anything but spawning. makes sense. still exciting stuff!

**Comment by Arnuh** [comment]
Considered complete in terms of what needs to be implemented in EchoPet. Any issues/feature requests related to the Model Engine pets should be created at the proper github page https://github.com/Arnuh/EchoPet-ModelEngine
